File: numerosPrimos.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

function isPrime(number) {
  if (number <= 1) {
    return false;
  }
  
  for (let i = 2; i <= Math.sqrt(number); i++) {
    if (number % i === 0) {
      return false;
    }
  }
  
  return true;
}

readline.question('Digite um número: ', (number) => {
  if (isPrime(number)) {
    console.log(`${number} é um número primo.`);
  } else {
    console.log(`${number} não é um número primo.`);
  }
  
  readline.close();
});

File: numerosPrimos.java
public class numerosPrimos {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Digite um número inteiro positivo: ");
        int num = sc.nextInt();
        boolean ehPrimo = true;

        for (int i = 2; i <= num/2; i++) {
            if (num % i == 0) {
                ehPrimo = false;
                break;
            }
        }
        
        if (num <= 1) {
            ehPrimo = false;
        }
        
        if (ehPrimo) {
            System.out.println(num + " é um número primo.");
        } else {
            System.out.println(num + " não é um número primo.");
        }
        
        sc.close();
    }

}
